src/services/pin/hooks/prepare-activity-log.js

```diff
@@ -77,14 +80,31 @@ const prepareActivityLog = () => (hook) => {
       };
       // Attach data for log-activity hook
       hook.data.logInfo = logInfo; // eslint-disable-line no-param-reassign
+      // For newly assigning (null -> assigned dept.), because the pin status will
+      // be changed, the notification will be handled by pin-transition service.
+      // For already assigned pin, if there is a change of assigned department,
+      // we need to inform both of departments' heads.
+      /* eslint-disable no-underscore-dangle,no-param-reassign */
+      const prevDeptId = pin.assigned_department ?
+        pin.assigned_department._id : '';
+      const newDeptId = updatedFieldObjects.assigned_department ?
+        updatedFieldObjects.assigned_department._id : '';
+      if (prevDeptId !== '' && newDeptId !== '' && prevDeptId !== newDeptId) {
```

Comment:
from no department to having department.
will be handled by state transition because it is firstly assigned.